core/session.rb

```diff
@@ -86,20 +88,23 @@ def setup(what)
 
           next if value['provider'] == "mdbci" # skip 'mdbci' block
           #
-          if value['box'].to_s =~ URI::regexp # THERE CAN BE DONE CUSTOM EXCEPTION
-            puts 'vagrant box add '+key.to_s+' '+value['box'].to_s
-            shell = 'vagrant box add '+key.to_s+' '+value['box'].to_s
-          else
-            puts 'vagrant box add --provider virtualbox '+value['box'].to_s
-            shell = 'vagrant box add --provider virtualbox '+value['box'].to_s
-          end
-
-          # TODO: resque Exeption
-          system shell # THERE CAN BE DONE CUSTOM EXCEPTION
-
-          exit_code = $?.exitstatus
-          possibly_failed_command = shell
-
+          begin
+	          if value['box'].to_s =~ URI::regexp # THERE CAN BE DONE CUSTOM EXCEPTION
+	            puts 'vagrant box add '+key.to_s+' '+value['box'].to_s
+	            shell = 'vagrant box add '+key.to_s+' '+value['box'].to_s
+	          else
+	            puts 'vagrant box add --provider virtualbox '+value['box'].to_s
+	            shell = 'vagrant box add --provider virtualbox '+value['box'].to_s
+	          end
+
+	          system shell # THERE CAN BE DONE CUSTOM EXCEPTION
+	          exit_code = $?.exitstatus
+	          raise "failed command: #{shell}" unless exit_code==0
+	      
+	      rescue Exception => error
```

Comment:
Your raise is caught with rescue and is not really thrown. Is it supposed to be so? Application will not return 1.